registry/remote/repository.go

```diff
@@ -895,7 +903,7 @@ func (s *manifestStore) Resolve(ctx context.Context, reference string) (ocispec.
 
 	switch resp.StatusCode {
 	case http.StatusOK:
-		return s.generateDescriptor(resp, ref)
+		return generateManifestDescriptor(resp, ref, req.Method, s.repo.MaxMetadataBytes)
```

Comment:
Looks like we can simplify this back to
```suggestion
		return s.generateDescriptor(resp, ref, req.Method)
```

Reply:
Yes; Done.